nntrainer/layers/rnncell.cpp

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 hyeonseok lee <hs89.lee@samsung.com>
+ *
+ * @file   rnncell.cpp
+ * @date   29 Oct 2021
+ * @brief  This is Recurrent Layer Cell Class of Neural Network
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author hyeonseok lee <hs89.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#include <cmath>
+
+#include <lazy_tensor.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+#include <rnncell.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+static constexpr size_t SINGLE_INOUT_IDX = 0;
+
+// - weight_xh ( weights of input to hidden )
+// - weight_hh ( weights of hidden to hidden )
+// - bias_h ( hidden bias )
+enum RNNCellParams { weight_xh, weight_hh, bias_h, hidden_state, dropout_mask };
+
+RNNCellLayer::RNNCellLayer() :
+  LayerImpl(),
+  rnncell_props(props::Unit(), props::HiddenStateActivation(), props::DropOutRate(),
+                props::MaxTimestep(), props::Timestep()),
+  wt_idx({0}),
+  acti_func(ActivationType::ACT_NONE, true),
+  epsilon(1e-3) {}
+
+void RNNCellLayer::finalize(InitLayerContext &context) {
```

Comment:
We need a check if num inputs == 1, within this function

Reply:
Already existed :) 
You can check it at line 56

Reply:
Right, wonder how I missed it :p